spec.html

```diff
@@ -34532,8 +34534,203 @@ <h2>Syntax</h2>
           [+UnicodeMode] `-`
           CharacterClassEscape[?UnicodeMode]
           CharacterEscape[?UnicodeMode]
+
+        ClassSetExpression ::
+          ClassUnion
+          ClassIntersection
+          ClassSubtraction
+
+        ClassUnion ::
+          ClassSetRange ClassUnion?
+          ClassSetOperand ClassUnion?
+
+        ClassIntersection ::
+          ClassSetOperand `&amp;&amp;` [lookahead != `&amp;`] ClassSetOperand
+          ClassIntersection `&amp;&amp;` [lookahead != `&amp;`] ClassSetOperand
+
+        ClassSubtraction ::
+          ClassSetOperand `--` ClassSetOperand
+          ClassSubtraction `--` ClassSetOperand
+
+        ClassSetRange ::
+          ClassSetCharacter `-` ClassSetCharacter
+      </emu-grammar>
+      <ul>
+        <li>
+          It is a Syntax Error if the CharacterValue of the first |ClassSetCharacter| is larger than the CharacterValue of the second |ClassSetCharacter|.
+        </li>
+      </ul>
+      <emu-grammar type="definition">
+        ClassSetOperand ::
+          ClassSetCharacter
+          ClassStringDisjunction
+          NestedClass
+
+        NestedClass ::
+          `[` [lookahead != `^`] ClassContents[+UnicodeMode, +UnicodeSetsMode] `]`
+          `[` `^` ClassContents[+UnicodeMode, +UnicodeSetsMode] `]`
+          `\` CharacterClassEscape[+UnicodeMode]
       </emu-grammar>
+      <emu-note>
+        <p>The first two lines here are equivalent to CharacterClass.</p>
+        <p style="background: yellow">[It is less confusing not having to back out of the new syntax, only to dive back in. And if we wanted to use CharacterClass here, we would need to check for differences in static &amp; runtime semantics.]</p>
+      </emu-note>
+      <emu-grammar type="definition">
+        ClassStringDisjunction ::
+          `\q{` ClassString MoreClassStrings? `}`
+
+        MoreClassStrings ::
+          `|` ClassString MoreClassStrings?
+
+        ClassString ::
+          [empty]
+          NonEmptyClassString
+
+        NonEmptyClassString ::
+          ClassSetCharacter NonEmptyClassString?
 
+        ClassSetCharacter ::
+          [lookahead &notin; ClassSetReservedDoublePunctuator] SourceCharacter but not ClassSetSyntaxCharacter
+          `\` CharacterEscape[+UnicodeMode]
+          `\` ClassSetReservedPunctuator
+          `\` `b`
+
+        ClassSetReservedDoublePunctuator :: one of
+          `&amp;&amp;` `!!` `##`
+          `$$` `%%` `**`
+          `++` `,,` `..`
+          `::` `;;` `&lt;&lt;`
+          `==` `&gt;&gt;` `??`
+          `@@` `^^` `__`
+          `&grave;&grave;` `~~`
```

Comment:
I opened https://github.com/tc39/proposal-regexp-set-notation/issues/59 about removing the special treatment of double punctuation, and https://github.com/tc39/proposal-regexp-set-notation/issues/60 about specifically excluding `.` and `_` from special reservation even if the former fails to pass muster.

Reply:
Closing the loop here. Sorry if there is already a resolving comment and this shows up in multiple places in the pull request view...

We discussed & closed the two issues. Since https://github.com/mathiasbynens/ecma262/pull/12 the underscore is no longer special.